model-optimizer/mo/main.py

```diff
@@ -139,8 +139,13 @@ def arguments_post_parsing(argv: argparse.Namespace):
         frameworks = ['tf', 'caffe', 'mxnet', 'kaldi', 'onnx']
         frameworks = list(set(frameworks + available_moc_front_ends))
         if argv.framework not in frameworks:
-            raise Error('Framework {} is not a valid target. Please use --framework with one from the list: {}. ' +
-                        refer_to_faq_msg(15), argv.framework, frameworks)
+            if argv.use_legacy_frontend:
+                raise Error('Framework {} is not a valid target when using the --use_legacy_frontend flag. '
+                            'Please use --framework with one from the list: {}. ' +
+                            refer_to_faq_msg(15), argv.framework, frameworks)
```

Comment:
I think it is still unclear for pdpd in case of use `--use_legacy_frontend` for pdpd. Because this message does not tell about that I shouldn't use `--use_legacy_frontend` for pdpd.
In your case it will be 'Please use --framework with one from the list: 'tf', 'caffe', 'mxnet', 'kaldi', 'onnx''.
because `get_moc_frontends(argv)` returns [] and

Reply:
argv.framework is None if I don't specify it explicitly.

Reply:
I adressed this by adding logic to guess the framework by model extension.

Reply:
Maybe in this message we can add 'available_moc_frontends' and have something like 'Available legacy frontends: tf, caffe, kaldi, onnx, Available non-legacy frontends: {available_moc_frontends}'